File: Src/Icm.Core/Repository/IRepositoryOfTEntityTKey.vb
Namespace Icm.Data

    Public Interface IRepository(Of TEntity, TKey)
        Inherits IEnumerable(Of TEntity)

        Sub Add(ByVal entity As TEntity)
        Function GetById(key As TKey) As TEntity

        Function Contains(entity As TEntity) As Boolean

        Function ContainsId(key As TKey) As Boolean

        Sub Delete(ByVal entity As TEntity)

    End Interface

End Namespace

File: Src/Icm.Core/Repository/IRepositoryOfTEntityTKey.cs
using System.Collections.Generic;

namespace Icm.Data
{

	public interface IRepository<TEntity, TKey> : IEnumerable<TEntity>
	{

		void Add(TEntity entity);
		TEntity GetById(TKey key);

		bool Contains(TEntity entity);

		bool ContainsId(TKey key);


		void Delete(TEntity entity);
	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
